tests/integrations/langchain/langchain_nv_ai_endpoints_test.py

```diff
@@ -0,0 +1,543 @@
+import os
+
+import pytest
+from langchain_core.messages import AIMessageChunk
+from langchain_nvidia_ai_endpoints import ChatNVIDIA
+
+import weave
+from weave.integrations.integration_utilities import op_name_from_ref
+
+model = "meta/llama-3.1-8b-instruct"
+
+
+@pytest.mark.skip_clickhouse_client  # TODO:VCR recording does not seem to allow us to make requests to the clickhouse db in non-recording mode
+@pytest.mark.vcr(
+    filter_headers=["authorization"],
+    allowed_hosts=["api.wandb.ai", "localhost"],
+)
+def test_chatnvidia_quickstart(client: weave.trace.weave_client.WeaveClient) -> None:
+    api_key = os.environ.get("NVIDIA_API_KEY", "DUMMY_API_KEY")
+
+    nvidia_client = ChatNVIDIA(
+        api_key=api_key, model=model, temperature=0.0, max_tokens=64, top_p=1
+    )
+
+    response = nvidia_client.invoke("Hello!")
+
+    calls = list(client.calls())
+    # need to make 2 because of the langchain integration getting a call in there
+    assert len(calls) == 2
+    call = calls[1]
+
+    assert response.content is not None
+
+    assert op_name_from_ref(call.op_name) == "langchain.Llm.ChatNVIDIA-generate"
+    assert call.started_at is not None
+    assert call.started_at < call.ended_at  # type: ignore
+
+    output = call.output
+    assert output["model"] == model
+    assert output["object"] == "chat.completion"
+
+    usage = call.summary["usage"][output["model"]]  # type: ignore
+    assert usage["requests"] == 1
+    assert usage["completion_tokens"] == 24
+    assert usage["prompt_tokens"] == 12
+    assert usage["total_tokens"] == 36
+
+    inputs = call.inputs
+    assert inputs["model"] == model
+    assert inputs["messages"] == [{"role": "user", "content": "Hello!"}]
+    assert inputs["max_tokens"] == 64
+    assert inputs["temperature"] == 0.0
+    assert inputs["top_p"] == 1
+
+
+@pytest.mark.skip_clickhouse_client  # TODO:VCR recording does not seem to allow us to make requests to the clickhouse db in non-recording mode
+@pytest.mark.vcr(
+    filter_headers=["authorization"], allowed_hosts=["api.wandb.ai", "localhost"]
+)
+@pytest.mark.asyncio
+async def test_chatnvidia_async_quickstart(
+    client: weave.trace.weave_client.WeaveClient,
+) -> None:
+    api_key = os.environ.get("NVIDIA_API_KEY", "DUMMY_API_KEY")
+
+    nvidia_client = ChatNVIDIA(
+        api_key=api_key, model=model, temperature=0.0, max_tokens=64, top_p=1
+    )
+
+    response = await nvidia_client.ainvoke("Hello!")
+
+    calls = list(client.calls())
+    # need to make 2 because of the langchain integration getting a call in there
+    assert len(calls) == 2
+    call = calls[1]
+
+    assert response.content is not None
+
+    assert op_name_from_ref(call.op_name) == "langchain.Llm.ChatNVIDIA-generate"
+    assert call.started_at is not None
+    assert call.started_at < call.ended_at  # type: ignore
```

Comment:
why do you need these type ignores?

Reply:
lmao thats leftover from tests copied from openai, will remove

Reply:
so do you still need this?